### PR TITLE
ui: query peer review frontend

### PR DESCRIPTION
**Pull request by zhangvi7**
## Context
-  Users need to enable this feature on desired query engine in admin settings
- When enabled query engine is selected, the request review option is shown to allow users to request and input reviewers for their query
- We are keeping the review on the execution, not on a query cell
 
## Changes
- Feature toggle on query engines to enable peer review option on a query cell
- Peer Review Modal 
   
## Test Plan

**Enabling query engine feature toggle**

image

**Requesting a review on an execution**

https://github.com/user-attachments/assets/dbf301c1-fe9a-44be-b036-a936ea54a321

**Updated Request Review Modal**

image

### CODE REVIEW

File: querybook/webapp/components/QueryPeerReviewModal/QueryPeerReviewModal.tsx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+                    externalRecipients,
+                    notifierName,
+                    values.justification,
+                    query,
+                    queryEngine
+                );
+                onHide();
+                toast.success(
+                    'Review request sent! Your reviewers have been notified, and if approved, your query will be executed.'
+                );
```
Comment (jczhong84) [inline querybook/webapp/components/QueryPeerReviewModal/QueryPeerReviewModal.tsx]: after sending the request, where can the user to check all the requested executions and their status?

File: querybook/webapp/components/QueryPeerReviewModal/QueryPeerReviewModal.tsx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+                                    <strong>here</strong>.
+                                </Link>
+                            </Message>
+                        </div>
+                        <Message
+                            className="mb12"
+                            type="warning"
+                            size="small"
+                            message="Tip: Before submitting your sensitive query for review, run it once to validate its syntax and feasibility. This helps identify errors early and reduces unnecessary review cycles."
+                        />
```
Comment (jczhong84) [inline querybook/webapp/components/QueryPeerReviewModal/QueryPeerReviewModal.tsx]: wanna put the tip into the public config as well?

File: querybook/webapp/components/QueryPeerReviewModal/QueryPeerReviewModal.tsx (hunk 1 above)
Comment (zhangvi7) [reply querybook/webapp/components/QueryPeerReviewModal/QueryPeerReviewModal.tsx]: Good point, I will need to create 2nd tab in Request Review Modal above to monitor existing all reviews. 
I think we can also add a link in the QueryExecutionBar to review. Will handle these in later PR, and also copy paste review link ability

File: querybook/webapp/components/QueryPeerReviewModal/QueryPeerReviewModal.tsx [hunk 3]
```diff
@@ -126,7 +129,7 @@ const QueryPeerReviewForm: React.FC<IQueryPeerReviewFormProps> = ({
                             className="mb12"
                             type="warning"
                             size="small"
-                            message="Tip: Before submitting your sensitive query for review, run it once to validate its syntax and feasibility. This helps identify errors early and reduces unnecessary review cycles."
+                            message={reviewTip}
```
Comment (jczhong84) [inline querybook/webapp/components/QueryPeerReviewModal/QueryPeerReviewModal.tsx]: I guess the review tip could be optional. only render the message banner when provided.

**Review by jczhong84 — APPROVED** [review]

**Pull request merged**